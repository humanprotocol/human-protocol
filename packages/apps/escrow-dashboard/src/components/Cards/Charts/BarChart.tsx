import { Box, Typography, useTheme } from '@mui/material';
import numeral from 'numeral';
import { FC } from 'react';
import {
  BarChart as RechartsBarChart,
  Bar,
  ResponsiveContainer,
  Tooltip,
  XAxis,
} from 'recharts';

import { Container } from '../Container';

type Series = {
  date: string;
  value: number;
};

type BarChartProps = {
  title: string;
  totalValue?: string | number;
  series: Series[];
};

export const BarChart: FC<BarChartProps> = ({ title, totalValue, series }) => {
  const theme = useTheme();

  return (
    <Container>
      <Typography variant="body2" color="primary" fontWeight={600} mb="4px">
        {title}
      </Typography>
      {totalValue !== undefined && (
        <Typography
          variant="h2"
          color="primary"
          sx={{ fontSize: { xs: 32, md: 48, lg: 64, xl: 80 } }}
        >
          {numeral(totalValue).format('0,0')}
        </Typography>
      )}
      <Box sx={{ width: '100%', height: 190 }}>
        <ResponsiveContainer>
          <RechartsBarChart
            data={series}
            margin={{ top: 30, left: 4, right: 4 }}
          >
            <XAxis
              dataKey="date"
              type="category"
              axisLine={false}
              tickLine={false}
              interval="preserveStartEnd"
              ticks={[
                series[0]?.date,
                series[Math.ceil(series.length / 2) - 1]?.date,
                series[series.length - 1]?.date,
              ]}
              tick={{ fill: '#320A8D', fontSize: '12px', fontFamily: 'Inter' }}
              tickMargin={12}
              padding={{ left: 10, right: 10 }}
            />
            <Tooltip />
            <Bar dataKey="value" fill={theme.palette.primary.main} />
          </RechartsBarChart>
        </ResponsiveContainer>
      </Box>
    </Container>
  );
};
