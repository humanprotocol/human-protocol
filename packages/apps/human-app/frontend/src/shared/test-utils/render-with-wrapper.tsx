import type { RenderOptions } from '@testing-library/react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import type { ReactElement, ReactNode } from 'react';

//custom wrapper from official documentation https://testing-library.com/docs/react-testing-library/setup/
export function renderWithWrapper(
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) {
  function Wrapper({ children }: { children: ReactNode }) {
    return <BrowserRouter>{children}</BrowserRouter>;
  }

  return render(ui, { wrapper: Wrapper, ...options });
}
