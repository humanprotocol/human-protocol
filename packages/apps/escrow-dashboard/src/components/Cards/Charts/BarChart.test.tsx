import { render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { create } from 'react-test-renderer';

import { BarChart } from './BarChart';

const mock = {
  title: 'Title',
  totalValue: 100,
  series: [
    { date: '23/09', value: 1 },
    { date: '24/09', value: 2 },
    { date: '25/09', value: 3 },
  ],
};

describe('when rendered BarChart component', () => {
  it('should render passed prop `title`', async () => {
    await act(() => {
      render(
        <BarChart
          title={mock.title}
          totalValue={mock.totalValue}
          series={mock.series}
        />
      );
    });
    expect(screen.findByLabelText(mock.title)).toBeTruthy();
  });

  it('should render passed prop `totalValue`', async () => {
    await act(() => {
      render(
        <BarChart
          title={mock.title}
          totalValue={mock.totalValue}
          series={mock.series}
        />
      );
    });
    expect(screen.findByLabelText(mock.totalValue)).toBeTruthy();
  });
});

it('BarChart component renders correctly, corresponds to the snapshot', () => {
  const tree = create(
    <BarChart
      title={mock.title}
      totalValue={mock.totalValue}
      series={mock.series}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
