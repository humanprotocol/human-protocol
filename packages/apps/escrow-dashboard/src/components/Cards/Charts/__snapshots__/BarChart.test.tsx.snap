// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BarChart component renders correctly, corresponds to the snapshot 1`] = `
<div
  className="MuiBox-root css-19n1ffs"
>
  <p
    className="MuiTypography-root MuiTypography-body2 css-dapz8w-MuiTypography-root"
  >
    Title
  </p>
  <h2
    className="MuiTypography-root MuiTypography-h2 css-1ag1ig1-MuiTypography-root"
  >
    100
  </h2>
  <div
    className="MuiBox-root css-zgm2l"
  >
    <div
      className="recharts-responsive-container"
      style={
        {
          "height": "100%",
          "maxHeight": undefined,
          "minHeight": undefined,
          "minWidth": 0,
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
